control-plane-operator/controllers/metrics/metricscontroller.go

```diff
@@ -0,0 +1,162 @@
+package metrics
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/transport"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	promapi "github.com/prometheus/client_golang/api"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+var (
+	ComponentLabel = "app"
+	NamespaceLabel = "namespace"
```

Comment:
This may be redundant given the metrics are assigned a `namespace` label during scrape, meaning this value ends up in the `exported_namespace` field as a conflict with a duplicate value